docs/standard/commandline/syntax.md

```diff
@@ -0,0 +1,577 @@
+
+```console
+git clean -f -d -x
+git clean -fdx
+```
+
+If an argument is provided after an option bundle, it applies to the last option in the bundle. The following command lines are equivalent:
+
+```console
+myapp -a -b -c arg
+myapp -abc arg
+```
+
+In both variants in this example, the argument `arg` would apply only to the option `-c`.
+
+## Boolean options (flags)
+
+If `true` or `false` is passed for an option having a `bool` argument, it's parsed as expected. But an option whose argument type is `bool` typically doesn't require an argument to be specified. Boolean options, sometimes called "flags", typically have an [arity](#argument-arity) of `ArgumentArity.ZeroOrOne`. The presence of the option name on the command line, with no argument following it, results in a default value of `true`. The absence of the option name in command-line input results in a value of `false`. If the `myapp` command prints out the value of a Boolean option named `--interactive`, the following input creates the following output:
+
+```console
+myapp
+myapp --interactive
+myapp --interactive false
+myapp --interactive true
+```
+
+```output
+False
+True
+False
+True
+```
+
+In the .NET CLI, this pattern doesn't always hold true. Some Boolean options result in the same behavior when you pass `false` as when you pass `true`. This behavior results when .NET CLI code that implements the option only checks for the presence or absence of the option, ignoring the value. An example is `--no-restore` for the `dotnet build` command. Pass `no-restore false` and the restore operation will be skipped the same as when you specify `no-restore true` or `no-restore`.
+
+## The --help option
+
+Command-line apps typically provide an option to display a brief description of the available commands, options, and arguments. `System.CommandLine` automatically generates help output. For example:
+
+```dotnetcli
+dotnet list --help
+```
+
+```output
+Description:
+  List references or packages of a .NET project.
+
+Usage:
+  dotnet [options] list [<PROJECT | SOLUTION>] [command]
+
+Arguments:
+  <PROJECT | SOLUTION>  The project or solution file to operate on. If a file is not specified, the command will search the current directory for one.
+
+Options:
+  -?, -h, --help  Show command line help.
+
+Commands:
+  package    List all package references of the project or solution.
+  reference  List all project-to-project references of the project.
+```
+
+App users might be accustomed to different ways to request help on different platforms, so apps built on `System.CommandLine` respond to many ways of requesting help. The following commands are all equivalent:
+
+```dotnetcli
```

Comment:
Do we have an easy way for nothing following the command to give help. Personally I think it is a bad idea, but it is not uncommon.

Reply:
>Do we have an easy way for nothing following the command to give help. Personally I think it is a bad idea, but it is not uncommon.

@jonsequitur ?